gengokit/gengokit.go

```diff
@@ -160,134 +265,64 @@ func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []stri
+	var svcFuncs []string
+	for _, meth := range meths {
+		svcFuncs = append(svcFuncs, meth.GetName())
 	}
-	serviceFunctions = append(serviceFunctions, "NewBasicService")
-
-	serverHandlerFile := getFileByName(packageName+serverHandlerFilePath, previousFiles)
-	clientHandlerFile := getFileByName(packageName+clientHandlerFilePath, previousFiles)
-
-	for _, templateFilePath := range templateFileAssets.AssetNames() {
-		if filepath.Ext(templateFilePath) != ".gotemplate" {
-			log.WithField("Template file", templateFilePath).Debug("Skipping rendering non-buildable partial template")
-			continue
-		}
-
-		var generatedFilePath string
-		var generatedCode *bytes.Buffer
-		var err error
-
-		if templateFilePath == "NAME"+serverHandlerFilePath+"template" && serverHandlerFile != nil {
-			// If there's an existing service file, update its contents
+	svcFuncs = append(svcFuncs, "NewService")
 
-			generatedCode, err = g.updateServerMethods(serverHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy service handler file")
-			}
-
-		} else if templateFilePath == "NAME"+clientHandlerFilePath+"template" && clientHandlerFile != nil {
-			// If there's an existing client_handler file, update its contents
-
-			generatedCode, err = g.updateClientMethods(clientHandlerFile, serviceFunctions)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "could not modifiy client handler file")
-			}
-
-		} else {
-
-			generatedCode, err = applyTemplateFromPath(templateFilePath, g.templateExec)
-			if err != nil {
-				return nil, errors.Wrap(err, "could not render template")
-			}
-		}
-
-		generatedFilePath = templateFilePath
-		// Change file path from .gotemplate to .go
-		generatedFilePath = strings.TrimSuffix(generatedFilePath, "template")
-
-		// Turn code buffer into string and format it
-		formattedCode := formatCode(generatedCode.Bytes())
-
-		var resp truss.SimpleFile
+	return svcFuncs
+}
 
-		// Switch "NAME" in path with packageName.
-		//i.e. for packageName = addsvc; /NAME-service/NAME-server -> /addsvc-service/addsvc-server
-		resp.Path = strings.Replace(generatedFilePath, "NAME", packageName, -1)
+// trimTemplateExecutorServiceFuncs removes functions in funcsInFile from the
+// passed templateExecutor and returns a pointer to that templateExecutor
+func trimTemplateExecutorServiceFuncs(te templateExecutor, funcsInFile map[string]bool) (*templateExecutor, error) {
+	var methodsToTemplate []*deftree.ServiceMethod
 
-		resp.Write(formattedCode)
+	// Create temporary templateExec with only the methods we want to append
+	// We must also dereference the templateExec's Service and change our newly created
+	// Service's pointer to it's messages to be methodsToTemplate
 
-		codeGenFiles = append(codeGenFiles, &resp)
-	}
+	tempService := *te.Service
+	tempService.Methods = methodsToTemplate
 
-	return codeGenFiles, nil
-}
+	te.Service = &tempService
```

Comment:
`te.Service.Methods` is a pointer to `methodsToTemplate`, so ranging over `t.Service.Methods` and modifying `methodsToTemplate` means you're ranging over the thing you're modifying, which is bad and will probably break things. Fix ur pointers, yo!